R/study_exclusions_panel.R

```diff
@@ -83,11 +83,37 @@ study_exclusions_panel_server <- function(id, data, treatment_df, reference_trea
       studies <- isolate(all_studies())
       return(studies[!studies %in% main_connected_data()$Study])
     })
+    
+    # Update the checkboxes to tick the excluded studies
+    observe({
+      # Select check boxes for disconnected studies
+      disconnected_indices <- which(all_studies() %in% main_subnetwork_exclusions())
+      selected = all_studies()[disconnected_indices]
+      shiny::updateCheckboxGroupInput(inputId = "exclusionbox", selected = selected)
+    }) |> bindEvent(main_subnetwork_exclusions())
 
-    # Update "selections" for the initial data to be those excluded by taking the subnetwork
+    # Update which studies can be selected from the sensitivity analysis by taking the initial data subnetwork
     observe({
-      selections(main_subnetwork_exclusions())
-    })
+      filtered_treatments <- FindAllTreatments(main_connected_data())
+      
+      lapply(
+        all_studies(),
+        function(study) {
+          index <- match(study, all_studies())
+          subElement <- glue::glue("#{session$ns('exclusionbox')} .checkbox:nth-child({index}) label")
+          
+          study_treatments <- FindAllTreatments(data()[data()$Study == study, ])
+          
+          if (any(study_treatments %in% filtered_treatments)) {
+            # The 0ms delay is required to disable studies disconnected within the initial uploaded data
+            shinyjs::delay(0, shinyjs::enable(selector = subElement))
```

Comment:
I assume there is not around the delay 0?

Reply:
It just doesn't work without it. I'm not entirely sure why, but it's not trivial to work out why.